src/Sarif.Viewer.VisualStudio/SarifExplorerWindow.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Runtime.InteropServices;
+using System.Windows;
+using Microsoft.Sarif.Viewer.ErrorList;
+using Microsoft.Sarif.Viewer.Tags;
+using Microsoft.Sarif.Viewer.Views;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.Sarif.Viewer.Models;
+
+namespace Microsoft.Sarif.Viewer
+{
+    /// <summary>
+    /// This class implements the tool window exposed by this package and hosts a user control.
+    /// </summary>
+    /// <remarks>
+    /// In Visual Studio tool windows are composed of a frame (implemented by the shell) and a pane,
+    /// usually implemented by the package implementer.
+    /// <para>
+    /// This class derives from the ToolWindowPane class provided from the MPF in order to use its
+    /// implementation of the IVsUIElementPane interface.
+    /// </para>
+    /// </remarks>
+    [Guid("ab561bcc-e01d-4781-8c2e-95a9170bfdd5")]
+    public class SarifExplorerWindow : ToolWindowPane, IToolWindow
+    {
+        private ITrackSelection _trackSelection;
+        private SelectionContainer _selectionContainer;
+        private ISarifErrorListEventSelectionService sarifErrorListEventSelectionService;
+        private ITextViewCaretListenerService<ITextMarkerTag> textViewCaretListenerService;
+
+        internal SarifToolWindowControl Control
```

Comment:
>SarifToolWindowControl [](start = 17, length = 22)

Should be `SarifExplorerWindowControl` now.

Reply:
Uh, yeah, but no... That would require changing the XAML markup... this change is already too big.

---
In reply to: [502554647](https://github.com/microsoft/sarif-visualstudio-extension/pull/214#discussion_r502554647) [](ancestors = 502554647)